### ISSUE TITLE
Sometimes appended file is reloaded completely instead of updated

### ISSUE THREAD

**Issue by petsuter** [zarunbal/LogExpert#16]
Usually when a log file is appended to, the only visible update is that the new lines are shown.

But sometimes (almost always with larger files?) the entire file seems to get reloaded (with a progress bar in the statusbar) and the line/text columns are removed and re-added. This is hugely distracting and slow and makes the program useless for such logfiles where this happens constantly.

**Comment by Hirogen** [comment]
This can be tested with continous written loglines (50-100+) per second in very large-Logfiles (800+ mb), that are opend via a network drive.

**Comment by zarunbal** [comment]
Thanks I'll try to reproduce this problem. 

@Hirogen does this behavior only happen after 800+ MB files? How big (characters and Encoding) are your lines (avg, just a rough estimation). Have you some special settings (highlighting, Actions, Buffer Sizes, ...)

**Comment by Pipecity** [comment]
I'm also seeing this behavior but on smaller files as well. I'm using default settings. 
Really frustrating.

**Comment by isosphere** [comment]
I've had this issue, or perhaps a different issue: A small file (~23 MB) would take awhile to load, then upon loading it would be reloaded from scratch. The result is that the file is just continuously reloading.

I've solved the problem in my particular case by selecting "Use legacy file reader" under "Memory/CPU".

**Comment by nicolas-raoul** [comment]
Same symptoms (text and "Line"/"Text" coluns reloading constantly) with a 200KB (yes very small) file updated every 5 seconds on Windows 10 Pro 1803 / LogExpert 1.6.13.

The workaround suggested at https://github.com/zarunbal/LogExpert/issues/16#issuecomment-428218551 solved the problem (now only the few new lines appear and the existing content moves smoothly).

**Comment by Hirogen** [comment]
> 
> Same symptoms (text and "Line"/"Text" coluns reloading constantly) with a 200KB (yes very small) file updated every 5 seconds on Windows 10 Pro 1803 / freshly installed LogExpert 1.6.13 with default settings.
> 

Can you reproduce the problem with the 200kb file all the time it is loaded?

**Comment by nicolas-raoul** [comment]
@Hirogen Yes I just switched `"Use legacy file reader"` off, closed LogExpert, reopened it, and the problem reproduced. Closed and reopened again, it reproduced again. The file is still under 1MB.

**Comment by Pr0metheus2** [comment]
I can reproduce same problem. Please see YT video. Could be problem linked to bug with multiple log files on network folder with extension .log, .log.1, .log.2 etc.?
https://youtu.be/eIlX-Jr2Tzc

**Comment by RandallFlagg** [comment]
@Hirogen 
This issue still exists. In order to reproduce it one needs to write more than one line before the thread does polling.

The code goes into the if instead of into the else:

```cs
if (newSize < FileSize || _isDeleted)
{
...
    if (!IsMultiFile)
    {
        OnLoadFile(new LoadFileEventArgs(_fileName, 0, true, _fileLength, true));
    }
...
}
else
{
...
    OnFileSizeChanged(args);
...
}
```

The solution will be to rewrite this part of the code and get rid of the thread and use [File System Watcher](https://learn.microsoft.com/en-us/dotnet/api/system.io.filesystemwatcher?view=net-9.0)

**Comment by Hirogen** [comment]
File System Watcher needs a thorough and precise implementation, otherwise it will be more of a problem then a solution.

https://weblogs.asp.net/ashben/31773

**Comment by RandallFlagg** [comment]
The only problem I see with FSW is that it doesn't work with sftp. Maybe a different implementation for each? I need to think about a good solution. I will try to do it in the next few weeks.

**Comment by margru** [comment]
I have the same problem. Once I installed version 1.20.0 I started to see this behaviour. Thus, I installed an older version (1.8.7) and I didn't observe this. However, now, suddenly, it started happenning even with this older version. It's unusable...

**Comment by Hirogen** [comment]
in newer versions there have been improvements, can you please retest if it still occurs?

**Comment by Hirogen** [comment]
could not longer reproduce with the new version and the SystemReaderDirect
